pandas/tests/io/excel/test_writers.py

```diff
@@ -1549,3 +1550,27 @@ def test_subclass_attr(klass):
     attrs_base = {name for name in dir(ExcelWriter) if not name.startswith("_")}
     attrs_klass = {name for name in dir(klass) if not name.startswith("_")}
     assert not attrs_base.symmetric_difference(attrs_klass)
+
+
+@pytest.mark.parametrize("merge_cells", [True, False])
+def test_excel_round_trip_with_periodindex(merge_cells, engine):
+    # GH#60099
+    df = DataFrame(
+        {"A": [1, 2]},
+        index=MultiIndex.from_arrays(
+            [period_range("2023-01", "2023-02", freq="M"), ["X", "Y"]],
+            names=["date", "category"],
+        ),
+    )
+
+    with BytesIO() as buffer:
+        with ExcelWriter(buffer) as writer:
+            df.to_excel(writer, merge_cells=merge_cells)
+
+        buffer.seek(0)
+        result_df = pd.read_excel(buffer, index_col=[0, 1])
+        result_df.index = result_df.index.set_levels(
+            [result_df.index.levels[0].to_period("M"), result_df.index.levels[1]]
+        )
```

Comment:
Instead of modifying the result, can you construct a DataFrame `expected` to match what `pd.read_excel(buffer, ...)` gives.

Reply:
Sure. I'll modify it. Thanks a lot!